### PR TITLE
Fixed hashing for Intercom secure_mode.

### PR DESCRIPTION
**Pull request by bbeaird**

### CODE REVIEW

File: integrations/intercom.php [hunk 1]
```diff
@@ -18,7 +18,7 @@ function segment_intercom_integration( $identify, $settings ) {
 		$identify['options'] = isset( $identify['options'] ) ? $identify['options'] : array();
 
 		$identify['options']['Intercom'] = array(
-			'userHash' => hash( 'sha256', $settings['use_intercom_secure_mode'] . $user_email )
+			'userHash' => hash( 'sha256', $settings['use_intercom_secure_mode'] . $user_id )
```
Comment (sperand-io) [inline integrations/intercom.php]: i dont see the $user_id variable in this scope?

**Comment by bbeaird** [comment]
Intercom says we generate the user_hash incorrectly. It needs to be userId, not email.

https://segment.zendesk.com/agent/tickets/42745

We say this in our docs too - https://segment.com/docs/integrations/intercom/#secure-mode

**Comment by bbeaird** [comment]
FWIW - per @tejasmanohar's comment from Slack, I think `$identify['user_id']` is actually email per this line - https://github.com/segmentio/analytics-wordpress/blob/7f9c0be6f2097acda34c6fa46b1b9a8548842494/analytics-wordpress.php#L741

File: integrations/intercom.php (hunk 1 above)
Comment (bbeaird) [inline integrations/intercom.php]: I assumed it was global scope (I'm guessing this was a bad assumption).

**Comment by sperand-io** [comment]
Throughout the wordpress plugin, we use the email *as* the userId (which is why the `$user_email` var in scope there is being assigned from `$identify['user_id'];` 

So I don't see how this is the issue. When we create the users in intercom we use the email as id, so the secure hash with the id (even if it's an email) should work just fine.

**Comment by bbeaird** [comment]
Ah...good call. I'll investigate more...

**Comment by bbeaird** [comment]
Intercom wants the hash to be of this form - Ruby code - `OpenSSL::HMAC.hexdigest('sha256', "123", "test@test.com")`. This produces the hash `76b3ea788c7046fc991a41d498ef65e6a303ba33aab2e33b764512fe442b1251`. With the updated PHP code, this `echo hash_hmac('sha256', "test@test.com", "123");` produces the same hash as above.

**Comment by tejasmanohar** [comment]
LGTM. What do we do about old data that was hashed differently? Do we need to add some sort of field that indicates the hashing algorithm used on the new data so it's clear there's a difference?

**Comment by bbeaird** [comment]
@tejasmanohar - I'm guessing this hasn't previously worked for anyone. Going forward, it should work if they input their app secret into the Wordpress plugin field.

**Comment by tejasmanohar** [comment]
:+1:

**Pull request merged**